plugins/generator-1.21.1/neoforge-1.21.1/templates/block/blockentity.java.ftl

```diff
@@ -218,6 +239,90 @@ public class ${name}BlockEntity extends RandomizableContainerBlockEntity impleme
 	}
     </#if>
 
+    <#if data.sensitiveToVibration>
+    @Override public VibrationSystem.Data getVibrationData() {
+    	return this.vibrationData;
+    }
+
+    @Override public VibrationSystem.User getVibrationUser() {
+    	return this.vibrationUser;
+    }
+
+    @Override public VibrationSystem.Listener getListener() {
+    	return this.vibrationListener;
+    }
+
+	private class VibrationUser implements VibrationSystem.User {
+
+		private final int x;
+		private final int y;
+		private final int z;
+		private final PositionSource positionSource;
+
+		public VibrationUser(BlockPos blockPos) {
+			this.x = blockPos.getX();
+			this.y = blockPos.getY();
+			this.z = blockPos.getZ();
+			this.positionSource = new BlockPositionSource(blockPos);
+		}
+
+		@Override public PositionSource getPositionSource() {
+			return this.positionSource;
+		}
+
+		<#if data.vibrationalEvents?has_content>
+		@Override public TagKey<GameEvent> getListenableEvents() {
+			return TagKey.create(Registries.GAME_EVENT, ResourceLocation.withDefaultNamespace("${data.getModElement().getRegistryName()}_can_listen"));
+		}
+		</#if>
+
+		@Override public int getListenerRadius() {
+			<#if hasProcedure(data.vibrationSensitivityRadius)>
+				Level world = ${name}BlockEntity.this.getLevel();
+				BlockState blockstate = ${name}BlockEntity.this.getBlockState();
+				return (int) <@procedureOBJToNumberCode data.vibrationSensitivityRadius/>;
+			<#else>
+				return ${data.vibrationSensitivityRadius.getFixedValue()};
+			</#if>
+		}
+
+		@Override public boolean canReceiveVibration(ServerLevel world, BlockPos vibrationPos, Holder<GameEvent> holder, GameEvent.Context context) {
+			<#if hasProcedure(data.canReceiveVibrationCondition)>
+			int vibrationX = vibrationPos.getX();

```

Comment:
You could also use an optimized procedure code here. See the example for return statements here: https://github.com/MCreator/MCreator/blob/8a249e55f3dccfc97670cffee8d24e276674ebe4/plugins/generator-1.21.1/neoforge-1.21.1/templates/block/blockentity.java.ftl#L159

However you need to handle default value in if here then, but this is also showcased in the linked example above.

Applies to both MC versions